GraphQL: Comparison list coverage (#8646)
magento/devdocs: src/guides/v2.4/graphql/mutations/delete-compare-list.md

---
group: graphql
title: deleteCompareList mutation
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

The `deleteCompareList` mutation deletes the specified comparison list.

> 

Reviewer: Can we emphasize to use this when the guest user signs out. Else there will be unnecessary records, I am not sure if there is a cron in place to clear those.

## Syntax

```graphql
mutation {
    deleteCompareList(
        uid: ID!
    ) {
        DeleteCompareListOutput
    }
}
```

## Example usage

> The following example deletes the comparison list that has a `uid` value of `sssXyGZkTFksdPnxNoK1ut6OiV4bbchD`.

Reviewer 2: ```suggestion
The following example deletes the comparison list with a `uid` value of `sssXyGZkTFksdPnxNoK1ut6OiV4bbchD`.
```

**Request:**

```graphql
mutation {
    deleteCompareList(
        uid: "sssXyGZkTFksdPnxNoK1ut6OiV4bbchD"
    ) {
        result
    }
}
```

**Response:**

```json
{
  "data": {
    "deleteCompareList": {
      "result": true
    }
  }
}
```

## Input attributes

The `deleteCompareList` mutation requires the following input.

Attribute |  Data Type | Description
--- | --- | ---
`uid` | ID! | The unique ID of the compare list to delete

## Output attributes

The `DeleteCompareListOutput` object returns a Boolean value that indicates whether the operation was successful.

Attribute |  Data Type | Description
--- | --- | ---

> `result` | Boolean! | A value of `true` indicates the compare list has been deleted successfully. Otherwise, the value is `false`

Reviewer 2: ```suggestion
`result` | Boolean! | A value of `true` indicates the compare list has been deleted successfully. Otherwise, `false`
```
